ton_client/src/proofs/mod.rs

```diff
@@ -192,8 +312,150 @@ pub async fn proof_block_data(
         compare_values(
             &params.block,
             &block_map.into(),
-            "blocks",
-            &COMPARE_JSON_IGNORE_FIELDS,
+            JsonPath::new("blocks"),
+            &COMPARE_BLOCK_JSON_IGNORE_FIELDS,
+            &COMPARE_BLOCK_JSON_NUMERIC_FIELDS,
+        )
+    {
+        return Err(Error::data_differs_from_proven(message));
+    }
+
+    Ok(())
+}
+
+#[derive(Serialize, Deserialize, Clone, ApiType, Default)]
+pub struct ParamsOfProofTransactionData {
+    /// Single transaction's data as queried from DApp server, without modifications.
+    /// The required fields are `id` and/or top-level `boc`, others are optional.
+    /// In order to reduce network requests count, it is recommended to provide `block_id` and `boc`
+    /// of transaction.
+    pub transaction: Value,
+}
+
+#[api_function]
+pub async fn proof_transaction_data(
+    context: Arc<ClientContext>,
+    params: ParamsOfProofTransactionData,
+) -> ClientResult<()> {
+    let engine = ProofHelperEngineImpl::new(context).await
+        .map_err(|err| Error::proof_check_failed(err))?;
+
+    let id_opt = params.transaction["id"].as_str();
+    let mut block_id_opt = params.transaction["block_id"].as_str()
```

Comment:
Can we use refrence instead of cloning the string?

Reply:
No, we also fill this value in `if` block later from the temporary query result.